ui/components/src/Table/hooks/useTableKeyboardNav.tsx

```diff
@@ -0,0 +1,126 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useCallback, useState } from 'react';
+
+export interface UseTableKeyboardNavProps {
```

Comment:
I pulled most of the keyboard interactions logic out into a hook to minimize how much it complicated the underlying table code. Also will hopefully make it easier to refactor in the future.